### ISSUE TITLE
[API] Change `fit` into `learn_graph` keyword

### ISSUE THREAD

**Issue by adam2392** [py-why/dodiscover#139]
Fit has a specific context in statistical learning, so instead, we should probably use a different keyword, `learn_graph`

**Comment by adam2392** [comment]
Additional changes I can think of are separating the PC/FCI/Psi-FCI rules into a set of static methods associated with the class. That way, they can be run on any graph, for example converting a DAG to PAG in pywhy-graphs.

```Python

# first infer separating sets using skeleton learner
sep_sets_ = learner.learn_graph(...)

# convert DAG to PAG without instantiating the FCI algorithm
pag = FCI.apply_orientation_rules(dag, sep_sets_)
```

**Comment by adam2392** [comment]
Closed by #141
